nasa/harmony
# Harmony 1420 reading polling queues

## Jira Issue ID
 HARMPONY-1420

## Description
This adds support for using service queues when trying to get work-items for services (work-polling). It only provides a temporary solution for filling the services queues with work-items for the sake of testing the reading functionality. The full solution will be completed in HARMONY-1419. The use of queues for work polling is disabled by default using a feature toggle.

This PR includes a refactoring that split the workflow-orchestration.ts file into multiple files concerned with work updates
and work polling, because the workfow-orchestration.ts file had gotten much too long. As such I would like to get this
merged as soon as possible as it will likely cause merge conflicts if we wait too long.

## Local Test Steps

1. check out this branch
2. run the tests to make sure they work with the feature toggle turned off
```
npm run test-fast
```
3. enable the feature toggle by adding `LOCALSTACK_HOST=localhost` and `USE_SERVICE_QUEUES=true` to your .env file
4. run the tests to make sure they pass with the queues
```
npm run test-fast
```
Optionally do a sandbox deployment to test the queues
5. set `USE_SERVICE_QUEUES=true` and `tf_create_service_queues=true` in your .env file in harmony-ci-cd
6. run `bin/deploy`
7. Run a query

Note that the loading of data onto service queues is kind of flaky right now and I have only tested it with synchronous 
queries. The point of this PR is to get the changes merged (without breaking anything) to avoid major merge conflicts.
I will not be closing HARMONY-1420 just yet.

## PR Acceptance Checklist
